pkg/grafana/sso_settings.go

```diff
@@ -0,0 +1,81 @@
+package grafana
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/grafana/grafana-openapi-client-go/client"
+	"github.com/grafana/grafana-openapi-client-go/models"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	grafanaAdminRole  = "Admin"
+	grafanaEditorRole = "Editor"
+	grafanaViewerRole = "Viewer"
+)
+
+func ConfigureSSOSettings(ctx context.Context, grafanaAPI *client.GrafanaHTTPAPI, organizations []Organization) error {
+	logger := log.FromContext(ctx)
+
+	provider := "generic_oauth"
+	resp, err := grafanaAPI.SsoSettings.GetProviderSettings(provider, nil)
+	if err != nil {
+		logger.Error(err, "failed to get sso provider settings.")
+		return errors.WithStack(err)
+	}
+
+	orgsMapping := generateGrafanaOrgsMapping(organizations)
+	settings := resp.Payload.Settings.(map[string]interface{})
+	settings["role_attribute_path"] = "to_string('Viewer')"
+	settings["org_attribute_path"] = "groups"
+	settings["org_mapping"] = fmt.Sprintf("'%s'", orgsMapping)
```

Comment:
Do we really need to quote this ?

Reply:
We needed to in the past, I'll try without

Reply:
Because we need to quote organizations that have spaces in their names

Reply:
quoting of organization names happens [here](https://github.com/giantswarm/observability-operator/pull/203/files/0c7b930728894f4e276d95b0d6663f1283e4f63c#diff-28f23d7eba0cb8874675e65b18043c3f33717e3e3d48f42dd8621afd6aa54d88R60) AFAIK

I would assume the grafana tooling then takes care from here.

Reply:
But fine, you'll figure this out while testing.

Reply:
This quotes each entries yes but we need to quote the whole list still

Reply:
Looks like it's causing issues